### PR TITLE
Improve virtual weave Pairing test round 1

### PR DESCRIPTION
**Pull request by yunhanw-google**
--Remove duplicate src/test-apps/happy/lib/WeaveDeviceMgr.py, use the
default WeaveDeviceMgr.py in weave
--Remove src/test-apps/happy/test-templates/WeaveBleCentral.py, make
weave ble test use apis from default WeaveDeviceMgr.py instead of CLI
tool
--Fix customized port for Service Directory in ToolCommonOptions.cpp and
update virtual weave pairing test.
--Add scalable support for weave pairing service test
--Initially merge ble test into happy/lib/WeaveDeviceManager.py, further refactor
needed.

### CODE REVIEW

File: src/test-apps/ToolCommonOptions.cpp [hunk 1]
```diff
@@ -823,7 +823,7 @@ WEAVE_ERROR ServiceDirClientOptions::GetRootDirectoryEntry(uint8_t *buf, uint16_
 
     Write8(buf, 0x41);                                       // Entry Type = Host/Port List, Host/Port List = 1
     LittleEndian::Write64(buf, kServiceEndpoint_Directory);  // Service Endpoint Id = Directory Service
-    Write8(buf, 0x80);                                       // Host ID Type = Fully Qualified, Suffix Index Present = false, Port Id Present = true
+    Write8(buf, 0x08);                                       // Host ID Type = Fully Qualified, Suffix Index Present = false, Port Id Present = true
```
Comment (robszewczyk) [inline src/test-apps/ToolCommonOptions.cpp]: We messed this up once, lets fix it for real this time: 
line 824:

    uint8_t hostPortListLength = 1;
    Write8(buf,  kDirectoryEntryType_HostPortList | (hostPortListLength & kMask_HostPortListLen));

line 826: 

     Write8(buf, kHostIdType_FullyQualified |  kMask_PortIdPresent /* | kMask_SuffixIndexPresent);

File: src/test-apps/happy/lib/WeaveDeviceManager.py [hunk 2]
```diff
@@ -32,7 +32,20 @@
 import base64
 import json
 import set_test_path
-import WeaveDeviceMgr
+
+try:
+    import WeaveDeviceMgr
+except Exception:
+    weaveDeviceMgrPath = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "..", "..", "..", "src", "device-manager", "python"))
```
Comment (robszewczyk) [inline src/test-apps/happy/lib/WeaveDeviceManager.py]: This seems really dodgy.  Is there a more readable way of finding an appropriate the paths?

File: src/test-apps/happy/lib/WeaveDeviceManager.py [hunk 3]
```diff
@@ -147,9 +162,38 @@ class ExtendedOption (Option):
...
+            bleManager = BleManager(devMgr)
+
+            try:
+                bleManager.ble_adapter_select(identifier=options.bleSrcAddr)
+            except WeaveDeviceMgr.DeviceManagerException, ex:
+                print ex
+                exit()
+
+            try:
+                line = ' -t 120 ' + str(options.bleDstAddr)
```
Comment (robszewczyk) [inline src/test-apps/happy/lib/WeaveDeviceManager.py]: Lets not do that.  I would really prefer we did not append random timeouts to the command line

File: src/test-apps/happy/lib/WeaveDeviceManager.py [hunk 4]
```diff
@@ -147,9 +162,38 @@ class ExtendedOption (Option):
...
+                bleManager.scan_connect(line)
+            except WeaveDeviceMgr.DeviceManagerException, ex:
+                print ex
+                exit()
+
             devMgr.ConnectBle(bleConnection=FAKE_CONN_OBJ_VALUE,
-                                   pairingCode=options.pairingCode,
-                                   accessToken=options.accessToken)
+                                       pairingCode=options.pairingCode,
```
Comment (robszewczyk) [inline src/test-apps/happy/lib/WeaveDeviceManager.py]: Weird indent, please fix

File: src/test-apps/happy/lib/WeaveDeviceManager.py (hunk 2 above)
Comment (yunhanw-google) [reply src/test-apps/happy/lib/WeaveDeviceManager.py]: it seems it is a little difficult on it.
Looking at the below path, in build folder, weave-device-mgr is linking to  /usr/local/openweave-core/src/device-manager/python/weave-device-mgr.py, and there is no WeaveDeviceManager folder, then it seems we need use WeaveDeviceManager from openweave-core/src/device-manager/python/WeaveDeviceManager.py.
build/x86_64-unknown-linux-gnu/src/device-manager/python$ ls -al weave-device-mgr
weave-device-mgr -> /usr/local/openweave-core/src/device-manager/python/weave-device-mgr.py

Further, the so file has been stored in .libs, it seems we need to add this so path via adding dmLibPath in system path.
openweave-core/build/x86_64-unknown-linux-gnu/src/device-manager/python/.libs$ ls
_WeaveDeviceMgr.a   _WeaveDeviceMgr.lai                                             _WeaveDeviceMgr_la-WeaveDeviceManager-BlePlatformDelegate.o  _WeaveDeviceMgr.so
_WeaveDeviceMgr.la  _WeaveDeviceMgr_la-WeaveDeviceManager-BleApplicationDelegate.o  _WeaveDeviceMgr_la-WeaveDeviceManager-ScriptBinding.o

File: src/test-apps/ToolCommonOptions.cpp (hunk 1 above)
Comment (yunhanw-google) [reply src/test-apps/ToolCommonOptions.cpp]: Done, thanks

File: src/test-apps/happy/lib/WeaveDeviceManager.py (hunk 3 above)
Comment (yunhanw-google) [reply src/test-apps/happy/lib/WeaveDeviceManager.py]: Done, thanks

File: src/test-apps/happy/lib/WeaveDeviceManager.py (hunk 4 above)
Comment (yunhanw-google) [reply src/test-apps/happy/lib/WeaveDeviceManager.py]: Done, thanks

File: src/test-apps/happy/lib/WeaveDeviceManager.py (hunk 2 above)
Comment (yunhanw-google) [reply src/test-apps/happy/lib/WeaveDeviceManager.py]: Done, move these paths in weave.py and make this a little better on readability in weave.py, and generate paths based upon self.weave_path

t

File: src/test-apps/happy/lib/WeaveDeviceManager.py [hunk 5]
```diff
@@ -32,7 +32,19 @@
 import base64
 import json
 import set_test_path
-import WeaveDeviceMgr
+
+
+weaveDeviceMgrPath = os.environ['weaveDeviceMgrPath']
```
Comment (robszewczyk) [inline src/test-apps/happy/lib/WeaveDeviceManager.py]: Change the variable name to be ALL_CAPS_WITH_UNDERSCORES

Comment (yunhanw-google) [reply src/test-apps/happy/lib/WeaveDeviceManager.py]: Done, thanks

**Pull request merged**